### PR TITLE
Bug 1955089: Add IPv6 check to support observe parallel tests

### PR DESCRIPTION
**Pull request by ardaguclu**
This PR adds IPv6 validation for `.spec.clusterIP` to be able to run observe tests in IPv6 environments . Because in IPv6 environments, clusterIP returns in format of `fd02::982f` instead of `172.30.0.1` and test `[sig-cli] oc observe works as expected [Suite:openshift/conformance/parallel]` is failing. 

Moreover, this PR adds `Sync ended` control for `oc observe daemonsets --once`.

### CODE REVIEW

File: test/extended/cli/observe.go
Comment (stbenjam) [inline test/extended/cli/observe.go:29]: It's not clear to me why we need this change here?

File: test/extended/cli/observe.go
Comment (stbenjam) [inline test/extended/cli/observe.go:71]: Could we include the full IPv6 address like they do for IPv4? I would guess it's predictable for IPv6 if it's predictable for IPv4

File: test/extended/cli/observe.go
Comment (ardaguclu) [reply test/extended/cli/observe.go:29]: In general case, `oc observe daemonsets --once` might successfully be synced and command can return `Sync ended` and this needs to be captured in test. For example, we have a daemonset, namely image-cache, and when we run this command, it returns `Sync ended` instead raising an error `Nothing to sync, exiting immediately`.

File: test/extended/cli/observe.go
Comment (ardaguclu) [reply test/extended/cli/observe.go:71]: Good point. Let me check if I can extract the IPv6.

Comment (ardaguclu) [reply test/extended/cli/observe.go:71]: Unfortunately, there is no statically defined IPv6. It seems that cluster is configured for IPv4;

`apiVersion: v1
kind: Service
metadata:
  labels:
    external: service
  name: external
  resourceVersion: "1"
  uid: 19cff995-5546-11e5-9f57-080027c5bfa9
spec:
  clusterIP: 172.30.0.100
  ports:
  - nodePort: 0
    port: 443
    protocol: TCP
    targetPort: 443
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}`

**Comment by ardaguclu** [comment]
https://github.com/openshift/origin/pull/25909 handles "Sync ended" problem and handling way of IPv6 might be different than just checking ":".

File: test/extended/cli/observe.go
Comment (stbenjam) [inline test/extended/cli/observe.go:71]: Does this work? I thought the cluster IP would be something like fd02::982f.

Comment (ardaguclu) [reply test/extended/cli/observe.go:71]: Actually I'm trying to find the most common specific IPv6 pattern for all. When I looked at the default kubernetes service;
`{
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
        "creationTimestamp": "2021-04-06T06:41:02Z",
        "labels": {
            "component": "apiserver",
            "provider": "kubernetes"
        },
        "name": "kubernetes",
        "namespace": "default",
        "resourceVersion": "231",
        "uid": "58f12afe-f051-4d8c-8af1-6c6285f53791"
    },
    "spec": {
        "clusterIP": "fd02::1",
        "clusterIPs": [
            "fd02::1"
        ],
        "ports": [
            {
                "name": "https",
                "port": 443,
                "protocol": "TCP",
                "targetPort": 6443
            }
        ],
        "sessionAffinity": "None",
        "type": "ClusterIP"
    },
    "status": {
        "loadBalancer": {}
    }
}`

The other option is like the previous commit checking `:`.

Comment (stbenjam) [reply test/extended/cli/observe.go:71]: We could hardcode `fd02::1` like they've hardcoded `172.30.0.1`.  

It appears to make an assumption that it's always the first address in the service network, so if that assumption is OK for IPv4 it should be OK for IPv6 too.

**Comment by andfasano** [comment]
/lgtm

@adambkaplan can you have please have a look? It's required for metal-ipi ipv6 jobs

**Review by adambkaplan — APPROVED** [review]
/lgtm

@stbenjam as we are past feature freeze, we need a Bugzilla issue to merge this test fix.

**Review by soltysh — APPROVED** [review]
/lgtm
from CLI pov

**Comment by stbenjam** [comment]
/hold

e2e-metal-ipi-ovn-ipv6 is currently broken from https://bugzilla.redhat.com/show_bug.cgi?id=1956281

No sense in retesting until the fix for that merges.

**Pull request merged**
